CHANGELOG.md

```diff
@@ -1,3 +1,23 @@
+## 2.15.0-rc.0 / 2019-12-18
+
+* [CHANGE] Discovery: Removed `prometheus_sd_kubernetes_cache_*` metrics. #6393
+* [FEATURE] API: Added new endpoint for exposing per metric metadata `/metadata`. #6420 #6442
+* [FEATURE] Remote write: Added config option to name remote target which is added as another label dimension to `prometheus_remote_storage_*` metrics. #6043
+* [FEATURE] Metrics format: Added examplar support to OpenMetrics parser. #6292
+* [FEATURE] Remote write: Added `prometheus_remote_storage_sent_bytes_total` counter per queue. #6344
+* [ENHANCEMENT] TSDB: Reduced memory footprint of loaded TSDB blocks. #6418 #6461
+* [ENHANCEMENT] TSDB: Significantly optimized what we buffer during compaction which should result in lower memory footprint during compaction. #6422 #6452 #6468 #6475
+* [ENHANCEMENT] promql: Improved query evaluation performance. #6427
+* [ENHANCEMENT] promql: Improved PromQL parser performance. #6356
+* [ENHANCEMENT] Mixin: Simplified remote-write dashboard in Prometheus mixin. #6335
+* [ENHANCEMENT] React UI: Implemented missing pages like `/targets`#6276, TSDB status page #6267 and many other fixes and performance improvements.
+* [ENHANCEMENT] TSDB: Improve replay latency. #6230
+* [BUGFIX] Config: Fixed alertmanager configuration to not miss targets when configurations are similar. #6455
+* [BUGFIX] Remote write: Value of `prometheus_remote_storage_shards_desired` gauge shows raw value of desired shards and it's updated correctly. #6378
+* [BUGFIX] Rules: Prometheus no longer accepts badly written rules and alerts that have same metric name and labelset. #6469
+* [BUGFIX] API: Targets Metadata API `/targets/metadata` accepts now empty `match_targets` parameter as in the spec. #6303
+* [BUGFIX] TSDB: WAL size is now used for size based retention calculation. #5886
```

Comment:
This isn't a bugfix, it's an enhancement.

Reply:
Really? Without this, the calculation was off.

Reply:
It was that way by design, not by accident. Improving it is an enhancement.

Reply:
Agreed offline that it's enhancement as we were aware of this